docs/docs/tutorial/creating-a-source-plugin/part-6/index.mdx

```diff
@@ -0,0 +1,621 @@
+      ({ actions }) => {
+        const { createTypes } = actions
+
+        createTypes(`
+          type ${NODE_TYPES.Post} implements Node {
+            id: ID!
+            _id: Int!
+            slug: String!
+            title: String!
+            author: ${NODE_TYPES.Author} @link(by: "name")
+            image: ${NODE_TYPES.Post}Image!
+          }
+
+          type ${NODE_TYPES.Author} implements Node {
+            id: ID!
+            _id: Int!
+            name: String!
+          }
+
+    // highlight-next-line
+          type ${NODE_TYPES.Asset} implements Node & RemoteFile {
+            url: String!
+            alt: String!
+            width: Int!
+            height: Int!
+          }
+        `)
+      }
+    ```
+
+    Nodes that `createAssetNode` creates will be GraphQL root nodes now.
+
+    <Collapsible
+      summary={<em>Type builder example</em>}
+    >
+
+    If you're not using the SDL syntax but type builders, you'd add it like this:
+
+    ```ts
+    schema.buildObjectType({
+      name: NODE_TYPES.Asset,
+      fields: {
+        url: `String!`,
+        alt: `String!`,
+        width: `Int!`,
+        height: `Int!`,
+      },
+      interfaces: [`Node`, `RemoteFile`],
+    })
+    ```
+
+    </Collapsible>
+
+1. Create a foreign-key relationship between `Post` and `Asset` through `@link` on the `image` field.
+
+    ```ts:title=plugin/src/create-schema-customization.ts
+    createTypes(`
+      type ${NODE_TYPES.Post} implements Node {
+        id: ID!
+        _id: Int!
+        slug: String!
+        title: String!
+        author: ${NODE_TYPES.Author} @link(by: "name")
+    // highlight-next-line
+        image: ${NODE_TYPES.Asset} @link
+      }
+
+      type ${NODE_TYPES.Author} implements Node {
+        id: ID!
+        _id: Int!
+        name: String!
+      }
+
+      type ${NODE_TYPES.Asset} implements Node & RemoteFile {
+        url: String!
+        alt: String!
+        width: Int!
+        height: Int!
+      }
+    `)
+    ```
+
+    If you remember from Part 3, the default behavior of `@link` is to use the `id` from the target node. Keep that in mind for the following instructions.
+
+1. Open the `plugin/src/source-nodes.ts` file and add a return statement to the `createAssetNode` function. Return the generated `id`:
+
+    ```ts:title=plugin/src/source-nodes.ts
+    export function createAssetNode(
+      gatsbyApi: SourceNodesArgs,
+      data: IPostImageInput
+    ) {
+      const id = gatsbyApi.createNodeId(`${NODE_TYPES.Asset}-${data.url}`)
+
+      const assetNode = {/* Contents */} satisfies IRemoteImageNodeInput
+
+      gatsbyApi.actions.createNode(assetNode)
+
+      // highlight-next-line
+      return id
+    }
+    ```
+
+1. A node of type `Post` should have the generated `id` of `Asset` as its `image` field. Because only posts can have images. You can conditionally add data to the node if certain contiditions are met like this:
```

Comment:
Small typo and suggestion to reword:
```suggestion
1. A node of type `Post` should have the generated `id` of `Asset` as its `image` field. Because only posts can have images, you can conditionally add data to the node if certain conditions are met like this:
```